process/executionTrack/executionResultsTracker.go

```diff
@@ -184,8 +184,11 @@ func (est *executionResultsTracker) cleanConfirmedExecutionResults(headerExecuti
 		}
 
 		executionResultFromTracker := pendingExecutionResult[idx]
-
-		areEqual := executionResultFromTracker.Equal(executionResultFromHeader)
+		sameHash := string(executionResultFromTracker.GetHeaderHash()) == string(executionResultFromHeader.GetHeaderHash())
+		sameNonce := executionResultFromTracker.GetHeaderNonce() == executionResultFromHeader.GetHeaderNonce()
+		sameRound := executionResultFromTracker.GetHeaderRound() == executionResultFromHeader.GetHeaderRound()
+		sameRootHash := string(executionResultFromTracker.GetRootHash()) == string(executionResultFromHeader.GetRootHash())
+		areEqual := sameHash && sameNonce && sameRound && sameRootHash
```

Comment:
So, we don't have the `Equal` function anymore? It looks like this code is begging to be extracted to a separate function.